src/main/scala/no/ndla/imageapi/service/ConverterService.scala

```diff
@@ -59,6 +60,52 @@ trait ConverterService {
         domainImageMetaInformation.captions.map(asApiCaption))
     }
 
+    def asApiImageMetaInformationWithSingleLanguage(domainImageMetaInformation: domain.ImageMetaInformation,
+                                                            language: String = AllLanguages): Option[api.ImageMetaInformationSingleLanguage] = {
+
+      val supportedLanguages = domainImageMetaInformation.titles.map(_.language.getOrElse(NoLanguage))
+        .++:(domainImageMetaInformation.alttexts.map(_.language.getOrElse(NoLanguage)))
+        .++:(domainImageMetaInformation.tags.map(_.language.getOrElse(NoLanguage)))
+        .++:(domainImageMetaInformation.captions.map(_.language.getOrElse(NoLanguage)))
+        .distinct
+        .filterNot(lang => lang.isEmpty)
+
+      val searchLanguage = if (language == AllLanguages) DefaultLanguage else language
```

Comment:
`all` / `AllLanguages` kommer fra kontrolleren, som sjekker om `language`-parameteret er satt. Ved å sende denne videre nedover så kan hver funksjon selv definere hva som skal skje dersom et spesifikt språk ikke er satt; som regel kommer dette automatisk til å bety `DefaultLanguage`, men f.eks. i article-api så finnes det unntak for dette.